src/app/profile/settings/page.tsx

```diff
@@ -1,8 +1,16 @@
-"use client";
 import UserSettings from "@/components/UserSettings";
+import getUserServerSide from "@/services/server/UserService";
 
-function ProfileSettings() {
-  return <UserSettings />;
+async function ProfileSettings() {
+  const user = await getUserServerSide();
+
+  return (
+    <>
+      {user !== undefined && (
+        <UserSettings fullName={user.full_name} email={user.email} />
+      )}
+    </>
+  );
```

Comment:
This is too complicated. We have a `React.Fragment` and also an `if` in JSX. We can do a normal `if` in code:

```
if (user !== undefined) {
  return ...
}

return null;
```

Also, please explain why we're not handling the situation where the user is undefined. I guess it's because the redirect happens already in the layout above, right?